Modules/posixmodule.c

```diff
@@ -6415,6 +6415,10 @@ os_sched_setaffinity_impl(PyObject *module, pid_t pid, PyObject *mask)
     }
     Py_CLEAR(iterator);
 
+    if (PyErr_Occurred()) {
```

Comment:
seems like this could be before the `Py_CLEAR`. Closer to the loop is better I think.

Reply:
I agree. Done.